### ISSUE TITLE
Nested `Type.Intersect` Errors

### ISSUE THREAD

**Issue by tristal** [sinclairzx81/typebox#887]
It seems that intersects are not propagating up errors from nested intersects when calling `Value.Errors(T, value)`.

For example:

```typescript
const A = Type.Intersect([
  Type.Object({
    id: Type.String(),
  }),
  Type.Object({
    x: Type.String(),
  }),
])
const B = Type.Intersect([
  Type.Object({
    id: Type.Number(),
  }),
  Type.Object({
    y: Type.String(),
  }),
])
const C = Type.Intersect([A, B])
```

Then doing `Value.Errors` call on both objects, yield:
```typescript
[...Errors(A, {})].map((r) => console.log(`> ${r.path}: ${r.message}`));
// > : Expected all values to match
// > /id: Required property
// > : Expected all values to match
// > /x: Required property

[...Errors(B, {})].map((r) => console.log(`> ${r.path}: ${r.message}`));
// > : Expected all values to match
// > /id: Required property
// > : Expected all values to match
// > /y: Required property

[...Errors(C, {})].map((r) => console.log(`> ${r.path}: ${r.message}`));
// > : Expected all values to match
// > : Expected all values to match
// > : Expected all values to match
// > : Expected all values to match
```

Is this expected?

**Comment by abdel** [comment]
This also appears to affect Unions as well, following on from your example:

```
// Discriminated Union example

const A = Type.Object({
	a: Type.String(),
	b: Type.Optional(Type.Never()),
	
	c: Type.Number()
});

const B = Type.Object({
	a: Type.Optional(Type.Never()),
	b: Type.String(),
	
	c: Type.Number()
});

const C = Type.Union([A, B]);
```

In version `0.30`, running `Value.Errors` on `A`, `B` and `C` gives the following:

```
# Value.Errors(A, {})
> /a: Expected string
> /c: Expected number
> /a: Expected required property
> /c: Expected required property

# Value.Errors(B, {})
> /b: Expected string
> /c: Expected number
> /b: Expected required property
> /c: Expected required property

# Value.Errors(C, {})
> : Expected value of union
> /a: Expected string
> /c: Expected number
> /a: Expected required property
> /c: Expected required property
> /b: Expected string
> /c: Expected number
> /b: Expected required property
> /c: Expected required property
```

In version `0.31` onwards, it instead gives the following output where the errors are no longer propagating:

```
# Value.Errors(A, {})
> /a: Required property
> /c: Required property
> /a: Expected string
> /c: Expected number

# Value.Errors(B, {})
> /b: Required property
> /c: Required property
> /b: Expected string
> /c: Expected number

# Value.Errors(C, {})
> : Expected union value
```

**Comment by sinclairzx81** [comment]
Hi @tristal, sorry for the delay getting to this. 

I have just published a update on 0.33.9 which fixes the immediate issue of not traversing through nested intersections. Revision 0.33.9 generates the following from your original repro.

```typescript
[...Value.Errors(A, {})].map((r) => console.log(`> ${r.path}: ${r.message}`));
// > /id: Expected required property
// > /id: Expected string
// > /x: Expected required property
// > /x: Expected string
// > : Expected all values to match

[...Value.Errors(B, {})].map((r) => console.log(`> ${r.path}: ${r.message}`));
// > /id: Expected required property
// > /id: Expected number
// > /y: Expected required property
// > /y: Expected string
// > : Expected all values to match

[...Value.Errors(C, {})].map((r) => console.log(`> ${r.path}: ${r.message}`));
// > /id: Expected required property
// > /id: Expected string
// > /x: Expected required property
// > /x: Expected string
// > : Expected all values to match  <-- interior intersection 1
// > /id: Expected required property
// > /id: Expected number
// > /y: Expected required property
// > /y: Expected string
// > : Expected all values to match <-- interior intersection 2
// > : Expected all values to match <-- exterior intersection
```
> Note, you can expect some duplicate errors for each intersection (noted above), this is mostly due to TypeBox error generation being stateless (and there is some expectation for downstream receivers of these errors to accumulate them based on the path), tho many of these details are yet to be ironed out.

This update applies to Intersect errors only. Union errors (also noted on this issue) unfortunately need to remain as they are until some additional infrastructure for TypeBox is written. In all, Error generation in TypeBox is an area that will be improved in later revisions, but generally open to thoughts and suggestions from the community on what they would like to see. Feedback welcome.

For now, this error has been closed out via PR. But happy to field any additional follow up discussions on this thread. 

Cheers!
S
